pkg/collector/corechecks/network-devices/cisco-sdwan/payload/interfaces.go

```diff
@@ -0,0 +1,78 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package payload
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/network-devices/cisco-sdwan/client"
+	devicemetadata "github.com/DataDog/datadog-agent/pkg/networkdevice/metadata"
+)
+
+// CiscoInterface is an interface to abstract the underlying network interface object (vEdge vs cEdge)
+type CiscoInterface interface {
+	// ID returns an unique interface ID
+	ID() string
+	// Index returns the interface index
+	Index() int
+	// Speed returns the interface speed
+	Speed() int
+	// OperStatus returns the interface oper status
+	OperStatus() devicemetadata.IfOperStatus
+	// AdminStatus returns the interface admin status
+	AdminStatus() devicemetadata.IfAdminStatus
+	// Metadata returns the interface metadata
+	Metadata(namespace string) devicemetadata.InterfaceMetadata
+	// IPAddressMetadata returns the metadata for this interface's IP addresses
+	IPAddressMetadata(namespace string) (devicemetadata.IPAddressMetadata, error)
+}
+
+// ProcessInterfaces process interfaces API payloads to build metadata and tags
+func ProcessInterfaces(namespace string, vEdgeInterfaces []client.InterfaceState, cEdgeInterfaces []client.CEdgeInterfaceState) (interfaces []devicemetadata.InterfaceMetadata, ipAddresses []devicemetadata.IPAddressMetadata, interfacesMap map[string]CiscoInterface) {
+	interfacesMap = make(map[string]CiscoInterface)
+
+	itfs := convertInterfaces(vEdgeInterfaces, cEdgeInterfaces)
+
+	for _, itf := range itfs {
+		_, present := interfacesMap[itf.ID()]
+
+		// Avoid sending duplicated interface metadata (In case the interface is returned both for IPv4 and IPv6)
```

Comment:
Good idea if this happens! What about the chance of "flipping" between ipv4 and ipv6 if the API returns stuff in a different order each time?
Should we prioritize one over the other?

Reply:
This happens for vEdge interfaces. It looks like IPv4 always comes first but we should probably not rely on that.
I'm assuming the metadata will be the same for the IPv4 and IPv6 interfaces (this might not be true for Oper/Admin states though), so I'm only using the first one. Do you think we should prioritise IPv4?

Reply:
I don't think there's any product reason to prioritize IPv4 or IPv6.
I see now that the metadata will be the same because ifName, and IPAddresses will be the same (both ip will be here)
So likely the metadata is more or less always the same.